File: src/main/java/org/openbase/planetsudo/view/level/levelobjects/MothershipPanel.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package org.openbase.planetsudo.view.level.levelobjects;

/*-
 * #%L
 * PlanetSudo GameEngine
 * %%
 * Copyright (C) 2009 - 2018 openbase.org
 * %%
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public
 * License along with this program.  If not, see
 * <http://www.gnu.org/licenses/gpl-3.0.html>.
 * #L%
 */

import org.openbase.planetsudo.game.GameObjectImages;
import java.awt.Graphics2D;
import java.awt.event.MouseEvent;
import org.openbase.planetsudo.level.levelobjects.Agent;
import org.openbase.planetsudo.level.levelobjects.Mothership;
import org.openbase.planetsudo.view.level.LevelPanel;

/**
 *
 * @author <a href="mailto:divine@openbase.org">Divine Threepwood</a>
 */
public class MothershipPanel extends AbstractLevelObjectPanel<Mothership, LevelPanel> {


	public MothershipPanel(Mothership resource, LevelPanel parentResourcePanel) {
		super(resource, resource.getPolygon(), GameObjectImages.Mothership.imagesURL, parentResourcePanel, DrawLayer.BACKGROUND);
		new TeamMarkerPanel(resource.getTeamMarker(), parentResourcePanel);
		loadAgentPanels();
	}

	private void loadAgentPanels() {
		for(Agent agent : resource.getAgents()) {
			new AgentPanel(agent, this);
		}
	}

	@Override
	protected void paintComponent(Graphics2D g2, Graphics2D gl) {
		boundingBox = resource.getBounds();
		//paintShape(g2);
		g2.setColor(resource.getTeam().getTeamColor());
		g2.fillRect((int)boundingBox.getCenterX()-45, (int)boundingBox.getCenterY()-45, 90,90);
		paintImage(g2);
		if(resource.isBurning()) {
			paintExplosion(g2);
		}
	}

	@Override
	public boolean isFocusable() {
		return false;
	}

	@Override
	protected void notifyMouseEntered() {
	}

	@Override
	protected void notifyMouseClicked(MouseEvent evt) {
	}
}

File: src/main/kotlin/org/openbase/planetsudo/view/level/levelobjects/MothershipPanel.kt
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package org.openbase.planetsudo.view.level.levelobjects

import org.openbase.planetsudo.game.GameObjectImages
import org.openbase.planetsudo.level.levelobjects.Mothership
import org.openbase.planetsudo.view.level.LevelPanel
import java.awt.Graphics2D

/**
 *
 * @author [Divine Threepwood](mailto:divine@openbase.org)
 */
class MothershipPanel(resource: Mothership, parentResourcePanel: LevelPanel) :
    AbstractLevelObjectPanel<Mothership, LevelPanel>(
        resource,
        resource.polygon,
        GameObjectImages.Mothership.imagesURL,
        parentResourcePanel,
        DrawLayer.BACKGROUND,
    ) {
    init {
        loadMarkerPanel()
        loadAgentPanels()
    }

    private fun loadMarkerPanel() {
        try {
            TeamMarkerPanel(resource.teamMarker, parentResourcePanel)
        } catch (ex: Throwable) {
            ex.printStackTrace()
        }
    }

    private fun loadAgentPanels() {
        for (agent in resource.getAgents()) {
            AgentPanel(agent, this)
        }
    }

    override fun paintComponent(g2: Graphics2D, gl: Graphics2D) {
        boundingBox = resource.bounds
        // paintShape(g2);
        g2.color = resource.team.teamColor
        g2.fillRect(boundingBox.centerX.toInt() - 45, boundingBox.centerY.toInt() - 45, 90, 90)
        paintImage(g2)
        if (resource.isBurning) {
            paintExplosion(g2)
        }
    }

    override val isFocusable: Boolean
        get() = false

    override fun notifyMouseEntered() {
    }
}
